packages/k2-alpine/src/components/SegmentedControl/SegmentedControl.tsx

```diff
@@ -28,86 +36,86 @@ export const SegmentedControl = ({
   dynamicItemWidth: boolean
   style?: ViewStyle
   type?: 'default' | 'thin'
-}): JSX.Element => {
+}): JSX.Element | null => {
+  const [isReady, setIsReady] = useState(true)
   const { theme } = useTheme()
-  const [viewWidth, setViewWidth] = useState<number>(0)
-  const [textWidths, setTextWidths] = useState<number[]>(
-    Array.from({ length: items.length }, () => 0)
-  )
-  const textRatios = useMemo(() => {
-    if (!dynamicItemWidth) {
```

Comment:
@onghwan I refactored the segmented control component to not rely on useState since these will happen on JS thread and the app is already so busy with the JS thread usually. it now uses shared values instead.